exporter/tanzuobservabilityexporter/metrics_test.go

```diff
@@ -35,21 +35,125 @@ func TestEndToEndGaugeConsumer(t *testing.T) {
+			Source: "my_source",
+		},
+	)
+
+	// Since internal metrics are coming from the exporter itself, we send
+	// them with an empty source which defaults to the host name of the
+	// exporter.
+	assert.Contains(
+		t,
+		sender.metrics,
+		tobsMetric{
+			Name:   missingValueMetricName,
+			Ts:     0,
+			Value:  0.0,
+			Tags:   typeIsGaugeTags,
+			Source: "",
+		},
+	)
+}
 
-	// Here we test what happens if the resourcetotelemetry consumer is turned on and is
-	// copying resource attributes such as "host.name" into the tags.
+func TestEndToEndGaugeConsumerWithResAttrsIncluded(t *testing.T) {
+	gauge := newMetric("gauge", pmetric.MetricTypeGauge)
+	dataPoints := gauge.Gauge().DataPoints()
+	dataPoints.EnsureCapacity(1)
+	exporterConfig := createDefaultConfig()
+	tobsConfig := exporterConfig.(*Config)
+	tobsConfig.Metrics.IncludeResourceAttrs = true
+	// Here we test what happens if IncludeResourceAttrs = true
 	addDataPoint(
 		432.25,
 		1640123456,
 		map[string]interface{}{"source": "renamed", "host.name": "my_source", "env": "prod"},
 		dataPoints,
 	)
-	resourceAttributes := map[string]string{"host.name": "my_source"}
+	resourceAttributes := map[string]string{"host.name": "my_source", "res_attr_key": "res_attr_value"}
 	metrics := constructMetricsWithTags(resourceAttributes, gauge)
 	sender := &mockGaugeSender{}
 	gaugeConsumer := newGaugeConsumer(sender, componenttest.NewNopTelemetrySettings())
 	consumer := newMetricsConsumer(
-		[]typedMetricConsumer{gaugeConsumer}, &mockFlushCloser{}, true)
+		[]typedMetricConsumer{gaugeConsumer}, &mockFlushCloser{}, true, tobsConfig.Metrics)
+	assert.NoError(t, consumer.Consume(context.Background(), metrics))
+
+	// The "host.name" tag gets filtered out as it contains our source, and the "source"
+	// tag gets renamed to "_source"
+	assert.Contains(
+		t,
+		sender.metrics,
+		tobsMetric{
+			Name:   "gauge",
+			Ts:     1640123456,
+			Value:  432.25,
+			Tags:   map[string]string{"_source": "renamed", "env": "prod", "res_attr_key": "res_attr_value"},
+			Source: "my_source",
+		},
+	)
+
+	// Since internal metrics are coming from the exporter itself, we send
+	// them with an empty source which defaults to the host name of the
+	// exporter.
+	assert.Contains(
+		t,
+		sender.metrics,
+		tobsMetric{
+			Name:   missingValueMetricName,
+			Ts:     0,
+			Value:  0.0,
+			Tags:   typeIsGaugeTags,
+			Source: "",
+		},
+	)
+}
+
+func TestEndToEndGaugeConsumerWithAppResAttrsExcluded(t *testing.T) {
+	gauge := newMetric("gauge", pmetric.MetricTypeGauge)
+	dataPoints := gauge.Gauge().DataPoints()
+	dataPoints.EnsureCapacity(1)
+	exporterConfig := createDefaultConfig()
+	tobsConfig := exporterConfig.(*Config)
+	tobsConfig.Metrics.ExcludeAppTags = true
+	// Here we test what happens if IncludeResourceAttrs = false ExcludeAppTags = false
```

Comment:
On line 143, it says ExcludeAppTags = true, but on this line it states that ExcludeAppTags = false. Which scenario did you intend to test?

Reply:
fixed the comment